internal/controller/deployment_controller.go

```diff
@@ -395,6 +395,12 @@ func (r *DeploymentReconciler) Delete(ctx context.Context, l logr.Logger, deploy
 		}
 		return ctrl.Result{}, err
 	}
+
+	err = r.keepOrReleaseAWSCluster(ctx, l, deployment.Namespace, deployment.Name)
```

Comment:
If we want this logic for AWS provider only, let's check the deployment type first (that infrastructure provider equals AWS). If we're going to keep this behavior for all providers, please adapt the name of the function and error messages to make it more common.